### ISSUE TITLE
Bug: `max-lines` rule also counting the eslint comments.

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#17830]
### Environment

Node version: latest

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint max-lines: ["error", 2]*/

var a, b, c;
```

### What did you expect to happen?

no errors

### What actually happened?

got error it says: File has too many lines (3). Maximum allowed is 2.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbWF4LWxpbmVzOiBbXCJlcnJvclwiLCAyXSovXG5cbnZhciBhLCBiLCBjOyIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by kecrily** [comment]
I'm not sure if this is the expected behavior

**Comment by Tanujkanti4441** [comment]
> I'm not sure if this is the expected behavior

@kecrily sorry due to exams and other stuffs wasn't able to respond.

i am too not sure if it is a bug as eslint comments to enable rule is also part of code, but it's documentation's examples are written in some ways that made me to open this issue. just wanna confirm so we can correct the docs or the rule as required

**Comment by mdjermanovic** [comment]
I think the current behavior is correct, and there's the option [`skipComments`](https://eslint.org/docs/latest/rules/max-lines#skipcomments) to not include comments in the count, but it seems like we should fix the examples.

**Comment by nzakas** [comment]
I agree that the current behavior is correct and we can update the docs.

**Comment by Tanujkanti4441** [comment]
Thanks for confirming i'll update the docs
